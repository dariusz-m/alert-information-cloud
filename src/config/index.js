import devConfig from "./config.development";
import prodConfig from "./config.production";
import testConfig from "./config.test";

const env = process.env.NODE_ENV || "development";

const config = {
    "development": devConfig,
    "production": prodConfig,
    "test": testConfig,
};

export default config[env];
