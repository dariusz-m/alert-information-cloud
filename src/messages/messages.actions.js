export const MessagesActionType = {
    SEND_MESSAGE: 'SEND_MESSAGE',
    MESSAGE_HAS_BEEN_SENT: 'MESSAGE_HAS_BEEN_SENT',
    MESSAGE_HAS_NOT_BEEN_SENT: 'MESSAGE_HAS_NOT_BEEN_SENT',
    SUBSCRIBE_MESSAGES_CHANNEL: 'SUBSCRIBE_MESSAGES_CHANNEL',
    MESSAGES_CHANNEL_HAS_BEEN_BROKEN: 'MESSAGES_CHANNEL_HAS_BEEN_BROKEN',
    MESSAGE_WAS_RECEIVED: 'MESSAGE_WAS_RECEIVED',
    CLEAR_ALL_THE_MESSAGES: 'CLEAR_ALL_THE_MESSAGES',
    GET_PREVIOUS_MESSAGES: 'GET_PREVIOUS_MESSAGES',
    PREVIOUS_MESSAGES_WERE_RECEIVED: 'PREVIOUS_MESSAGES_WERE_RECEIVED',
    SOMETHING_WENT_WRONG_TRYING_GET_PREVIOUS_MESSAGES: 'SOMETHING_WENT_WRONG_TRYING_GET_PREVIOUS_MESSAGES',
};

export const sendMessage = (message, userIds) => {
    return {
        type: MessagesActionType.SEND_MESSAGE,
        payload: {
            message,
            userIds
        }
    };
};

export const messageHasBeenSent = () => {
    return {
        type: MessagesActionType.MESSAGE_HAS_BEEN_SENT,
    };
};

export const messageHasNotBeenSent = () => {
    return {
        type: MessagesActionType.MESSAGE_HAS_NOT_BEEN_SENT,
    };
};

export const subscribeMessagesChannel = () => {
    return {
        type: MessagesActionType.SUBSCRIBE_MESSAGES_CHANNEL,
    };
};

export const messagesChannelHasBeenBroken = () => {
    return {
        type: MessagesActionType.MESSAGES_CHANNEL_HAS_BEEN_BROKEN,
    };
};

export const messageWasReceived = (messageData) => {
    return {
        type: MessagesActionType.MESSAGE_WAS_RECEIVED,
        payload: messageData
    };
};

export const clearAllTheMessages = () => {
    return {
        type: MessagesActionType.CLEAR_ALL_THE_MESSAGES,
    };
};

export const getPreviousMessages = () => {
    return {
        type: MessagesActionType.GET_PREVIOUS_MESSAGES,
    };
};

export const previousMessagesWereReceived = (messages) => {
    return {
        type: MessagesActionType.PREVIOUS_MESSAGES_WERE_RECEIVED,
        payload: messages,
    };
};

export const somethingWentWrongTryingGetPreviousMessages = () => {
    return {
        type: MessagesActionType.SOMETHING_WENT_WRONG_TRYING_GET_PREVIOUS_MESSAGES,
    };
};

