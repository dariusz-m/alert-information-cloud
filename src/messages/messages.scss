.messages {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  .selected-message {
    flex: 0 1 50%;
  }
}


.data-piece{
  display: flex;
  margin: 13px 0;
}

.result-block{
  position: relative;
  min-height: 100px;
  padding: 21px;
  margin-top: 21px;
  display: block;
  text-decoration: none;
  cursor: pointer;
}

.result-block-hash{

}

a{
  text-decoration: none;
}

a:active,
a:focus,
a:visited{
  color: inherit;
}

.result-block:before{
  content: '';
  position: absolute;
  display: block;
  left: 0; top: 0;
  width: 100%; height: 100%;
  background: linear-gradient(to right, rgba(255,255,255,1) 0%,rgba(255,255,255,0) 100%);
  transform: skewY(-3deg);
  transform-origin: left;
  opacity: 0.8;
  animation: animFloat3d 2s var(--acc-sine) infinite alternate;
}


.result-block:hover:before{
  opacity: 1;
  transition: opacity 100ms var(--acc-sine);
}

@keyframes animFloat3d{
  0%{
    transform: skewY(-3deg);
  }
  100%{
    transform: rotateY(21deg) rotateX(8deg) skewY(-3deg);
  }
}

@keyframes animFloat3dAlt{
  0%{
    transform: rotateZ(180deg) skewY(-3deg);
  }
  100%{
    transform: rotateZ(180deg) rotateX(10deg) skewY(-3deg);
  }
}

.result-block:nth-of-type(odd):before{
  transform: rotateZ(180deg) skewY(-3deg);
  transform-origin: center;
  animation: animFloat3dAlt 2s var(--acc-sine) infinite alternate;
}

.result-block .link{
  padding: 0;
 font-size: 14px;
}

.result-block .link:hover + .icon.href{
  transform: translateX(3px);
}

.result-block > *{
  position: relative;
}

.result-block .balance{
  text-align: left;
  margin-right: 13px;
}

.icon{
  display: inline-flex;
  align-items: center;
  padding-right: 5px;
  transition: transform 100ms var(--acc-sine);
}

.result-title-side{
  font-size: 13px;
  margin-left: 5px;
}

.icon.flow.from:before,
.icon.flow.to:before{
  content: '';
  width: 15px; height: 15px;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='15' height='15' viewBox='0 0 15 15'%3E%3Cdefs%3E%3Cstyle%3E .cls-1 %7B fill-rule: evenodd; fill: %231fa600; %7D %3C/style%3E%3C/defs%3E%3Cpath id='_copy' data-name='î¥† copy' class='cls-1' d='M5.609,1.266a0.15,0.15,0,0,0-.109.047l-0.094.094L2.063,4.734a0.759,0.759,0,0,0-.141.242,0.7,0.7,0,0,0-.047.227,0.7,0.7,0,0,0,.047.227,0.754,0.754,0,0,0,.141.242,0.642,0.642,0,0,0,.938,0l2.2-2.2V9.875a3.27,3.27,0,0,0,2.031,3.063,3.208,3.208,0,0,0,1.3.266H13.2a0.671,0.671,0,0,0,.484-0.18,0.649,0.649,0,0,0,.188-0.492,0.63,0.63,0,0,0-.187-0.477,0.672,0.672,0,0,0-.484-0.18H8.531a1.974,1.974,0,0,1-2-2V3.469l2.2,2.2a0.663,0.663,0,1,0,.938-0.937L6.344,1.406a0.748,0.748,0,0,0-.109-0.07,0.367,0.367,0,0,1-.094-0.07,1.034,1.034,0,0,0-.3-0.047,0.522,0.522,0,0,0-.234.047h0Z'/%3E%3C/svg%3E%0A");
}

.link{
  padding: 13px 21px 0 21px;
  font-size: 15px;
  cursor: pointer;
  display: flex;
}

.link:hover:after{
  transform: translateX(10px);
}

.truncate{
  white-space: nowrap;
  text-overflow: ellipsis;
  width: 34%;
  display: block;
  overflow: hidden
}

.truncate.abit{
  width: 89%;
}

.scrollable-wrapper {
  height: 400px;
  overflow: auto;
  width: 50%;
}
