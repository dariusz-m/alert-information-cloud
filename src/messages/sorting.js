export const ASCENDING = 1;
export const DESCENDING = 2;

export const sortItemsByKey = function(elements, key, order) {
    const sorted = Object.assign([], elements).sort((a, b) => {
        if (a[key] === b[key]) { return 0; }
        return a[key] > b[key] ? 1 : -1;
    });
    if (order === DESCENDING) {
        sorted.reverse();
    }
    return sorted;
};
