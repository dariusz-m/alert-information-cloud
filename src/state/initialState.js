const initialState = {
    router: {"location": null},
    userAccount: {
        loginPage: {
            errorMessage: "",
        },
        registerPage: {
            isUserRegistered: false,
            errorMessage: "",
        }

    },
    users: [],
    alertInformationPage: {
        messages: [],
        lastMessage: null,
        messageHasBeenSent: null
    }
};

export default initialState;
