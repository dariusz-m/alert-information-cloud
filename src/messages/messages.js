import "./messages.scss";

import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";
import {bindActionCreators} from "redux";
import ReactList from "react-list";
import TimeAgo from "react-timeago";

import MultiSelect from "../multiselect/multiSelect";
import {sendMessage, subscribeMessagesChannel} from "./messages.actions";
import {MessageNotification} from "../notification/notification";
import {logout} from "../user-account/user-account.actions";

/**
 * Main component, handling messages
 */
export class Messages extends React.Component {
    /**
     * Constructor.
     *
     * @param {Object} props: Component configuration.
     */
    constructor(props) {
        super(props);

        this.state = {
            selectedUsersIds: [],
            message: "",
            selectedMessage: null
        };

        this.onSelectedOptionsChange = this.onSelectedOptionsChange.bind(this);
        this.onMessageBodyChange = this.onMessageBodyChange.bind(this);
        this.sendMessage = this.sendMessage.bind(this);
        this.renderMessage = this.renderMessage.bind(this);
        this.displayMessageDetails = this.displayMessageDetails.bind(this);
    }

    /**
     * On selected options change
     *
     * @param {Array} selectedUsersIds: Selected users ids
     */
    onSelectedOptionsChange(selectedUsersIds) {
        this.setState({
            selectedUsersIds: selectedUsersIds
        });
    }

    /**
     * On message body change.
     *
     * @param {Event} changeEvent: Change event.
     */
    onMessageBodyChange(changeEvent) {
        this.setState({message: changeEvent.target.value});
    }

    /**
     * On message body change.
     *
     * @param {Event} clickEvent: Click event.
     */
    sendMessage(clickEvent) {
        if (this.state.selectedUsersIds.length > 0) {
            this.props.actions.sendMessage(this.state.message, this.state.selectedUsersIds);
        }
        clickEvent.preventDefault();
    }

    /**
     * Component did mount.
     */
    componentDidMount() {
        this.props.actions.subscribeMessagesChannel();
    }

    /**
     * Display message details
     *
     * @param {Object} message: Message
     * */
    displayMessageDetails(message) {
        this.setState({selectedMessage: message});
    }

    /**
     * Render component.
     *
     * @param {number} index: Index of message
     *
     * @returns {React.Element}
     */
    renderMessage(index) {
        return (
            <div
                key={this.props.messages[index].timestamp}
                className="result-block"
                onClick={this.displayMessageDetails.bind(this, this.props.messages[index])}
            >
                <div className={`data-piece`}>
                    <span className="result-title-side">
                        <TimeAgo date={this.props.messages[index].timestamp}/>
                    </span>
                </div>
                <div className={`data-piece`}>
                    <span className={`icon flow from`} id="icon-direction"/>
                    <a className="link result-attr-2"> {this.props.messages[index].from}</a>
                    <span className="icon href"/>
                </div>
                <div className={`data-piece`}>
                    <a className="link result-attr-3 truncate">
                        <span className="icon hashtag"/>
                        {this.props.messages[index].message}</a>
                    <span className="icon href"/>
                </div>
            </div>

        );
    }

    /**
     * Render component.
     *
     * @returns {React.Element}
     */
    render() {
        const options = this.props.users.map((user) => {
            return {id: user.id, text: user.email};
        });

        return (
            <div>
                <button className={"btn btn-danger float-right"} onClick={this.props.actions.logout}>Logout</button>
                <div className={"md-12 messages"}>
                    <div className="scrollable-wrapper">
                        <ReactList
                            itemRenderer={this.renderMessage}
                            length={this.props.messages.length}
                            pageSize={1}
                        />
                    </div>
                    {this.state.selectedMessage ?
                        <div className={"selected-message"}>
                            <h4><strong>{this.state.selectedMessage.from}</strong> -- sent you message</h4>
                            <p>{this.state.selectedMessage.message}</p>
                            <span>{new Date(this.state.selectedMessage.timestamp).toString()}</span>
                        </div>
                        : null
                    }
                </div>
                <div className={"md-12"}>
                    <MultiSelect
                        options={options}
                        selectedOptionsIds={this.state.selectedUsersIds}
                        updateSelectedOptions={this.onSelectedOptionsChange}
                    />
                    <div className="input-group mb-3">
                        <input
                            type="text" className="form-control" placeholder="Message body..."
                            aria-label="Message body..." aria-describedby="basic-addon2"
                            value={this.state.message} onChange={this.onMessageBodyChange}
                        />
                        <div className="input-group-append">
                            <button
                                className="btn btn-outline-secondary"
                                type="button"
                                disabled={!this.state.message}
                                onClick={this.sendMessage}
                            >
                                Send
                            </button>
                        </div>
                    </div>
                    <div className="md-12">
                        {(() => {
                            switch(this.props.messageHasBeenSent) {
                                case false:
                                    return(
                                        <div className="alert alert-danger">
                                            <div>
                                                <strong>Something went wrong!</strong> Try again.
                                            </div>
                                        </div>
                                    );
                                case true:
                                    return(
                                        <div className="alert alert-success">
                                            <strong>Success!</strong> Message has been sent.
                                        </div>
                                    );
                                default:
                                    return this.state.selectedUsersIds.length === 0 ?
                                        <div className="alert alert-info">
                                            <p>Select at least one recipient to send message.</p>
                                        </div>
                                        : null;

                            }
                        })()}
                    </div>
                </div>
                <MessageNotification message={this.props.lastMessage}/>
            </div>
        );
    }
}


Messages.propTypes = {
    users: PropTypes.arrayOf(PropTypes.shape({
        id: PropTypes.number.isRequired,
        email: PropTypes.string.isRequired,
        firstname: PropTypes.string.isRequired,
        lastname: PropTypes.string.isRequired,
    })).isRequired,
    messages: PropTypes.arrayOf(PropTypes.shape({
        timestamp: PropTypes.number.isRequired,
        message: PropTypes.string.isRequired,
        from: PropTypes.string.isRequired,
    })).isRequired,
    lastMessage: PropTypes.shape({
        timestamp: PropTypes.number.isRequired,
        message: PropTypes.string.isRequired,
        from: PropTypes.string.isRequired,
    }),
    messageHasBeenSent: PropTypes.bool,
    actions: PropTypes.shape({
        sendMessage: PropTypes.func.isRequired,
        subscribeMessagesChannel: PropTypes.func.isRequired
    }),
};

/**
 * Map state to props.
 *
 * @param {Object} state: State of application
 *
 * @returns {Object}
 */
function mapStateToProps(state) {
    return {
        users: state.users,
        messages: state.alertInformationPage.messages,
        lastMessage: state.alertInformationPage.lastMessage,
        messageHasBeenSent: state.alertInformationPage.messageHasBeenSent,
    };
}

/**
 * Map state to props.
 *
 * @param {Object} dispatch: Dispatch
 *
 * @returns {Object}
 */
function mapDispatchToProps(dispatch) {
    return {
        actions: bindActionCreators({
            sendMessage: sendMessage,
            subscribeMessagesChannel: subscribeMessagesChannel,
            logout: logout,
        }, dispatch)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(Messages);
