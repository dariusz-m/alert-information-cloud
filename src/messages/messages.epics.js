import {catchError, mapTo, mergeMap, retryWhen, map, switchMap, tap, pluck} from "rxjs/operators";
import {from, of, timer} from "rxjs";
import {ajax} from "rxjs/ajax";
import { webSocket } from 'rxjs/webSocket';

import config from "../config";
import {
    messageHasBeenSent,
    messageHasNotBeenSent,
    MessagesActionType,
    messagesChannelHasBeenBroken,
    messageWasReceived,
    previousMessagesWereReceived,
    somethingWentWrongTryingGetPreviousMessages
} from "./messages.actions";
import {decrypt} from "../security";
import {DESCENDING, sortItemsByKey} from "./sorting";

export const sendMessage = action$ =>
    action$.ofType(MessagesActionType.SEND_MESSAGE)
        .pipe(
            mergeMap((action) =>
                from(
                    ajax({
                        url: `${config.API_ORIGIN}/api/v1/message`,
                        method: "POST",
                        headers: {
                            'Content-Type': 'application/json',
                            'Authorization': sessionStorage.getItem("sessionId")
                        },
                        crossDomain: true,
                        withCredentials: false,
                        body: {
                            message: action.payload.message,
                            user_ids: action.payload.userIds // eslint-disable-line camelcase
                        }
                    })
                ).pipe(
                    mapTo(messageHasBeenSent("Message has been sent.")),
                    catchError(() => {
                        return of(messageHasNotBeenSent("Something went wrong, try again!"));
                    })
                )
            ),
        );

export const getPreviousMessages = action$ =>
    action$.ofType(MessagesActionType.GET_PREVIOUS_MESSAGES)
        .pipe(
            mergeMap(() =>
                from(
                    ajax({
                        url: `${config.API_ORIGIN}/api/v1/message`,
                        method: "GET",
                        headers: {
                            'Content-Type': 'application/json',
                            'Authorization': sessionStorage.getItem("sessionId")
                        },
                        crossDomain: true,
                        withCredentials: false,
                    })
                ).pipe(
                    pluck("response"),
                    map(messagesData => messagesData.map((messageData) => {
                        return {
                            message: messageData.message,
                            from: messageData.from_user.email,
                            timestamp: messageData.timestamp
                        };
                    })),
                    map((messages) => sortItemsByKey(messages, "timestamp", DESCENDING)),
                    map((sortedMessages) => previousMessagesWereReceived(sortedMessages)),
                    catchError((error) => {
                        console.error(error);
                        return of(somethingWentWrongTryingGetPreviousMessages());
                    })
                )
            ),
        );

export const subscribeMessagesChannel = action$ => {
    let socket$ = null;
    return action$.ofType(MessagesActionType.SUBSCRIBE_MESSAGES_CHANNEL)
        .pipe(
            tap(() => {
                socket$ = webSocket({
                    url: `wss://${config.SOCKET_HOSTNAME}/event-emitter?sid=${sessionStorage.getItem("sessionId")}`,
                    deserializer: function (e) { return e.data; },
                    serializer: function (value) { return value; },
                });
            }),
            switchMap(() =>
                socket$.multiplex(
                    () => {},
                    () => {},
                    () => true
                ).pipe(
                    retryWhen(genericRetryStrategy()),
                    map((encryptedData) => {
                        const decryptedData = JSON.parse(decrypt(
                            sessionStorage.getItem("clientPrivateKey"),
                            sessionStorage.getItem("serverPublicKey"),
                            encryptedData
                        ));
                        return messageWasReceived({
                            message: decryptedData.message,
                            from: decryptedData.from_user.email,
                            timestamp: decryptedData.timestamp
                        });
                    }),
                )
            ),
            catchError(() => {
                return of(messagesChannelHasBeenBroken());
            })
        );
};

const genericRetryStrategy = ({
    maxRetryAttempts = 5,
    scalingDuration = 1000,
    excludedStatusCodes = []
} = {}) => (attempts) => {
    return attempts.pipe(
        mergeMap((error, i) => {
            const retryAttempt = i + 1;
            // if maximum number of retries have been met
            // or response is a status code we don't wish to retry, throw error
            if (retryAttempt > maxRetryAttempts || excludedStatusCodes.find(e => e === error.status)) {
                throw(error);
            }
            console.error(`Attempt ${retryAttempt}: retrying in ${retryAttempt * scalingDuration}ms`);
            // retry after 1s, 2s, etc...
            return timer(retryAttempt * scalingDuration);
        }),
    );
};
