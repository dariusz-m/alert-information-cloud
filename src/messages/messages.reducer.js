import initialState from '../state/initialState';
import {MessagesActionType} from './messages.actions';

export default (state = initialState.alertInformationPage, action) => {
    switch (action.type) {
        case MessagesActionType.MESSAGE_WAS_RECEIVED:
            return {...state, messages: [action.payload, ...state.messages], lastMessage: action.payload};
        case MessagesActionType.PREVIOUS_MESSAGES_WERE_RECEIVED:
            return {...state, messages: [...state.messages, ...action.payload]};
        case MessagesActionType.SEND_MESSAGE:
            return {...state, messageHasBeenSent: null};
        case MessagesActionType.MESSAGE_HAS_BEEN_SENT:
            return {...state, messageHasBeenSent: true};
        case MessagesActionType.MESSAGE_HAS_NOT_BEEN_SENT:
            return {...state, messageHasBeenSent: false};
        case MessagesActionType.CLEAR_ALL_THE_MESSAGES:
            return {...initialState.alertInformationPage};
        default:
            return state;
    }
};
