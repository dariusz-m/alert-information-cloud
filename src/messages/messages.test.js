import React from 'react';
import nock from "nock";
import renderer from "react-test-renderer";

import configureStore from "../state";
import config from "../config";
import initialState from "../state/initialState";
import {getPreviousMessages, sendMessage} from "./messages.actions";
import {Messages} from "./messages";


jest.mock('react-list', () => 'div');


describe('Message', () => {
    let store;
    const keys = {
        publicKey: "024eaa6162f2244ff94d8d99009c956eb8c1031087c571e6c6364eeb1cd2c231a3",
        privateKey: "eea1de62230a3f119202a5bc7be4dd9f7fd19c4f83b1c957305db8be18f7b9d7"
    };

    beforeEach(() => {
        store = configureStore();
    });

    it('Messages component renders correctly', () => {
        const users = [
            {
                "id": 45,
                "email": "emial1@ids.pl",
                "firstname": "Qwertyuiop123$",
                "lastname": "Qwertyuiop123$",
                "active": true,
                "createdAt": 1528311326582,
                "lastLoginAt": null
            },
            {
                "id": 42,
                "email": "fdsfsdfs@o2.pl",
                "firstname": "Qwertyuiop123$",
                "lastname": "Qwertyuiop123$",
                "active": false,
                "createdAt": 1528397427078,
                "lastLoginAt": null
            }
        ];
        const messages = [
            {
                timestamp: Date.now(),
                message: "Message body",
                from: "from@o2.pl"
            },
            {
                timestamp: Date.now(),
                message: "Message body 3",
                from: "from3@o2.pl"
            }
        ];
        const actions = {
            sendMessage: jest.fn(),
            subscribeMessagesChannel: jest.fn(),
        };
        const tree = renderer.create(
            <Messages actions={actions} users={users} messages={messages}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Message has been sent', (done) => {
        expect.hasAssertions();
        sessionStorage.setItem("sessionId", "sessionId");
        sessionStorage.setItem("clientPrivateKey", keys.privateKey);
        sessionStorage.setItem("serverPublicKey", keys.publicKey);
        nock(config.API_ORIGIN)
            .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
            .post(`/api/v1/message`)
            .reply(202, {});
        const expectedState = {
            ...initialState,
            alertInformationPage: {...initialState.alertInformationPage, messageHasBeenSent: true}
        };
        const userIds = [0, 1];

        store.dispatch(sendMessage("Message body", userIds));

        store.subscribe(() => {
            expect(store.getState()).toEqual(expectedState);
            done();
        });
    });


    it('Get previous message', (done) => {
        expect.hasAssertions();
        const responseMessages = [
            {
                "message":"dfdfdfd",
                "from_user":{"id":1,"firstname":"User","lastname":"User","email":"aic2.project@wp.pl"},
                "timestamp":5
            },
            {
                "message":"fvf",
                "from_user":{"id":1,"firstname":"User","lastname":"User","email":"aic2.project@wp.pl"},
                "timestamp":4
            },
            {
                "message":"fvf",
                "from_user":{"id":1,"firstname":"User","lastname":"User","email":"aic2.project@wp.pl"},
                "timestamp":3
            },
            {
                "message":"dfdfd",
                "from_user":{"id":1,"firstname":"User","lastname":"User","email":"aic2.project@wp.pl"},
                "timestamp":2
            },
            {
                "message":"fdfdfd",
                "from_user":{"id":1,"firstname":"User","lastname":"User","email":"aic2.project@wp.pl"},
                "timestamp":1
            }
        ];
        const expectedMessages = responseMessages.map((messageData) => {
            return {
                message: messageData.message,
                from: messageData.from_user.email,
                timestamp: messageData.timestamp
            };
        });
        nock(config.API_ORIGIN)
            .defaultReplyHeaders({ 'access-control-allow-origin': '*' })
            .get(`/api/v1/message`)
            .reply(200, responseMessages);
        const expectedState = {
            ...initialState,
            alertInformationPage: {...initialState.alertInformationPage, messages: expectedMessages}
        };

        store.dispatch(getPreviousMessages());

        store.subscribe(() => {
            expect(store.getState()).toEqual(expectedState);
            done();
        });
    });
});
