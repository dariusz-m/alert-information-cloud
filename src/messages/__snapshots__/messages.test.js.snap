// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Messages component renders correctly 1`] = `
<div>
  <button
    className="btn btn-danger float-right"
  >
    Logout
  </button>
  <div
    className="md-12 messages"
  >
    <div
      className="scrollable-wrapper"
    >
      <div
        itemRenderer={[Function]}
        length={2}
        pageSize={1}
      />
    </div>
  </div>
  <div
    className="md-12"
  >
    <div
      className="multi-select"
    >
      <div
        className="options"
      >
        <div>
          <header>
            AVAILABLE USERS
          </header>
          <input
            className="filter-input"
            onChange={[Function]}
            placeholder="Search..."
          />
          <div
            className="select-all-visible"
          >
            <a
              onClick={[Function]}
            >
              Send to all
            </a>
          </div>
          <ul
            className="ms-list"
          >
            <li
              className=""
              onClick={[Function]}
            >
              emial1@ids.pl
            </li>
            <li
              className=""
              onClick={[Function]}
            >
              fdsfsdfs@o2.pl
            </li>
          </ul>
        </div>
      </div>
      <div
        className="selected-options"
      >
        <div>
          <header>
            SEND TO
          </header>
          <input
            className="filter-input"
            onChange={[Function]}
            placeholder="Search..."
          />
          
          <ul
            className="ms-list"
          />
        </div>
      </div>
    </div>
    <div
      className="input-group mb-3"
    >
      <input
        aria-describedby="basic-addon2"
        aria-label="Message body..."
        className="form-control"
        onChange={[Function]}
        placeholder="Message body..."
        type="text"
        value=""
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-outline-secondary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Send
        </button>
      </div>
    </div>
    <div
      className="md-12"
    >
      <div
        className="alert alert-info"
      >
        <p>
          Select at least one recipient to send message.
        </p>
      </div>
    </div>
  </div>
  <div>
    <input
      name="dummy-for-react-web-notification"
      style={
        Object {
          "display": "none",
        }
      }
      type="hidden"
    />
    <audio
      id="sound"
      preload="auto"
    >
      <source
        src="https://georgeosddev.github.io/react-web-notification/example/sound.mp3"
        type="audio/mpeg"
      />
      <source
        src="https://georgeosddev.github.io/react-web-notification/example/sound.ogg"
        type="audio/ogg"
      />
      <embed
        autostart="false"
        hidden="true"
        loop="false"
        src="https://georgeosddev.github.io/react-web-notification/example/sound.mp3"
      />
    </audio>
  </div>
</div>
`;
